src/bin/tikv-server.rs

```diff
@@ -589,6 +595,42 @@ fn get_store_path(matches: &Matches, config: &toml::Value) -> String {
     format!("{}", absolute_path.display())
 }
 
+fn start_server<T, S>(mut server: Server<T, S>, mut el: EventLoop<Server<T, S>>)
+    where T: RaftStoreRouter,
+          S: StoreAddrResolver + Send + 'static
+{
+    let ch = server.get_sendch();
+    let h = thread::Builder::new()
+        .name("tikv-server".to_owned())
+        .spawn(move || {
+            server.run(&mut el).unwrap();
+        })
+        .unwrap();
+    handle_signal(ch);
+    h.join().unwrap();
+}
+
+#[cfg(unix)]
+fn handle_signal(ch: SendCh<Msg>) {
+    use signal::trap::Trap;
+    use nix::sys::signal::{SIGTERM, SIGINT};
+    let trap = Trap::trap(&[SIGTERM, SIGINT]);
```

Comment:
Couldn't find any doc or example how trap should be used.
Will it rerun this piece of signal catching code if the specified signal happens? Like a callback.